packages/sitecore-jss-nextjs/src/components/ComponentPropsContext.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { createContext, useContext, ReactNode } from 'react';
+import { ComponentProps } from "../sharedTypes/component-props";
+
+/**
+ * Component props context which we are using in order to store data fetched on components level (getStaticProps/getServerSideProps)
+ */
+export const ComponentPropsReactContext = createContext<ComponentProps>({});
+
+/**
+ * Hook in order to get access to props related to specific component. Data comes from ComponentPropsContext.
+ * @see ComponentPropsContext
+ */
+export function useComponentProps<ComponentData>(componentName: string): ComponentData {
```

Comment:
Given my comment below about needing to handle multiple components with the same name - this will likely need to change to a componentUid parameter / key.

Reply:
Changed